### ISSUE TITLE
Regridding CMIP5 data

### ISSUE THREAD

**Issue by DamienIrving** [JiaweiZhuang/xESMF#49]
I'm regridding CMIP5 ocean temperature data from a curvilinear to rectilinear grid. Following the [relevant example](https://xesmf.readthedocs.io/en/latest/Curvilinear_grid.html) in the documentation, I've tried to used xESMF for this task without success:
https://github.com/DamienIrving/ocean-analysis/blob/master/development/xESMF_test.ipynb

I was wondering if there's an obvious mistake I've made in that notebook?

**Comment by JiaweiZhuang** [comment]
The code looks alright; it should be either ESMPy install issues or weird singularities in the input grid. Do you have an example data file (say just one time slice)?

**Comment by DamienIrving** [comment]
Yep. No problem - you can download a cut down version of the data file (first timestep only) [here](https://drive.google.com/file/d/1BCSJGoU8W1CSzXoqILlL-BFvYuXsaH41/view?usp=sharing).

**Comment by DamienIrving** [comment]
GitHub seems to be having issues rendering anything but the initial version of the notebook, so best to use this link: https://github.com/DamienIrving/ocean-analysis/blob/c9d46e50b3171c0000c43a8628b42dba4d9a881f/development/xESMF_test.ipynb

**Comment by JiaweiZhuang** [comment]
The problem is that the grid you pass to xESMF contains multiple disconnected tiles, although they are stored in a single 2D array:

![image](https://user-images.githubusercontent.com/25473287/55174757-65e32b00-5154-11e9-9fd3-f2c4549c59a2.png)

So, apparently, xESMF/ESMPy has trouble understanding the connectivity.

The error is fixed by breaking your full grid to several well-defined 2D tiles.

See this GitHub gist for full code:
https://nbviewer.jupyter.org/gist/JiaweiZhuang/7a620ccea7d686891fed0221f386a50b

A very similar issue is https://github.com/JiaweiZhuang/xESMF/issues/14#issuecomment-370175606

A more general discussion on multi-tile support is at #21.

**Comment by DamienIrving** [comment]
Thanks, @JiaweiZhuang. This is super helpful.

Interestingly, iris ([regrid_weighted_curvilinear_to_rectilinear](https://scitools.org.uk/iris/docs/latest/iris/iris/experimental/regrid.html?highlight=regrid_weighted_curvilinear_to_rectilinear#iris.experimental.regrid.regrid_weighted_curvilinear_to_rectilinear)) had similar problems with this particular model/grid, but [cdo remapbil](https://code.mpimet.mpg.de/projects/cdo/embedded/index.html#x1-5870002.12.1) handled the regirdding just fine.

**Comment by JiaweiZhuang** [comment]
Yes Iris exactly uses ESMPy under the hood so should have the same issue. CDO seems to use its own, non-ESMF regridding implementation.
